zipkin-storage/zipkin2_cassandra/src/main/java/zipkin2/storage/cassandra/DefaultSessionFactory.java

```diff
@@ -58,6 +58,7 @@
       if (cassandra.ensureSchema()) {
         session = closer.register(cluster.connect());
         Schema.ensureExists(cassandra.keyspace(), cassandra.searchEnabled(), session);
+        Schema.ensureExists(DEFAULT_KEYSPACE + "_udts", false, session);
```

Comment:
Can I suggest you to add more debug logs on key places? It was really hard for me to find cassandra.ensureSchema was really set to false, I thought it could be an error to set the variable... I think it's worth logging it as well, as debug

Not on this PR, but I think it would be great as well to have a command line option on zipkin server to dump all spring config. It's not something hard to do, but extremely helpful for the user.

Reply:
Spring endpoints already exist for these things.. noted here https://github.com/openzipkin/zipkin/blob/master/zipkin-server/README.md#endpoints

for example, if you hit /autoconfig you'd see if ensureSchema was set to false or not. If I weren't distracted while you were troubleshooting, I would probably have mentioned to use that. there's also /configprops

https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-endpoints.html

I use both of these a lot.. while we actively decided to not replicate spring docs, it might make sense to note both of these in the server README.

It is true that there aren't many debug statements in the storage code. Will add a couple.